classes/submission/maps/Schema.php

```diff
@@ -722,4 +727,69 @@ protected function appSpecificProps(): array
     {
         return [];
     }
+
+    /**
+     * Implement by a child class to get available editorial decisions data for a stage of a submission.
+     */
+    protected function getAvailableEditorialDecisions(int $stageId, Submission $submission): array
+    {
+        return [];
+    }
+
+    /**
+     * Check if a user can make Decisions or Recommendations on a submission's stage
+    */
+    protected function checkDecisionPermissions(int $stageId, Submission $submission, User $user, int $contextId): array
```

Comment:
For reference, see https://github.com/pkp/pkp-lib/blob/9e75633e4809dc79ce26b8e8a3287073478020bb/pages/workflow/PKPWorkflowHandler.php#L557-L607